### ISSUE TITLE
Use docker to enable CI builds

### ISSUE THREAD

**Issue by HybridEidolon** [rust3ds/ctru-rs-old#25]
No idea what this entails, but it might be useful to have CI builds on PRs.

**Comment by FenrirWolf** [comment]
Fixed by https://github.com/rust3ds/ctru-rs/pull/35

**Comment by HybridEidolon** [comment]
Not quite, unfortunately. This change (adding alloc_system to Cargo.toml) breaks the build according to Travis-CI. https://github.com/rust3ds/ctru-rs/commit/54d7fc6c88d8e884ed95bdefc3493b09e4e43573

**Comment by panicbit** [comment]
It builds fine here, maybe retry the build (you can do that from within travis).

**Comment by panicbit** [comment]
Wtf, it works fine locally and it breaks on travis. Devkit probably has nothing to do with it since it also built fine with the corresponding env vars unset.

**Comment by panicbit** [comment]
Oh, actually it's `cargo test` that is broken!

**Comment by FenrirWolf** [comment]
Things seem to be working pretty well now. One suggestion I have though is that the compile-test should separately report success and failure for builds against the pinned nightly vs the latest nightly. If you check out my recent PRs, you'll see that the test reports all green even though there's actually breakage on the latest nightly. Ideally it should say 1 passed, 1 failed

**Comment by panicbit** [comment]
@FenrirWolf If you'd prefer it that way, then just remove the `matrix` part from the `.travis.yml`.

**Comment by FenrirWolf** [comment]
I just figure that way we get a better heads-up about upcoming nightly breakage. Then when we fix it, we can re-pin a newer nightly, then wash, rinse, repeat.

Though if the idea is to not do frequent updates to the pinned nightly version, we can probably things the way they are. Just depends on what the goal is I suppose.
